apps/web/src/pages/integrations/IntegrationsStoreModal.tsx

```diff
@@ -0,0 +1,339 @@
+import { useEffect, useState } from 'react';
+import styled from '@emotion/styled';
+import {
+  ChannelTypeEnum,
+  IConfigCredentials,
+  ILogoFileName,
+  providers,
+  PushProviderIdEnum,
+  EmailProviderIdEnum,
+  InAppProviderIdEnum,
+} from '@novu/shared';
+import { Grid, Group, Modal, ActionIcon, useMantineColorScheme, createStyles, MantineTheme } from '@mantine/core';
+import * as cloneDeep from 'lodash.clonedeep';
+import { useIntegrations } from '../../hooks';
+import { When } from '../../components/utils/When';
+import { NovuEmailProviderModal } from './components/NovuEmailProviderModal';
+import { NovuInAppProviderModal } from './components/NovuInAppProviderModal';
+import { ChannelGroup } from './components/Modal/ChannelGroup';
+import { colors, shadows, Title } from '../../design-system';
+import { ConnectIntegrationForm } from './components/Modal/ConnectIntegrationForm';
+import { Close } from '../../design-system/icons/actions/Close';
+
+export function IntegrationsStoreModal({
+  openIntegration,
+  closeIntegration,
+}: {
+  openIntegration: boolean;
+  closeIntegration: () => void;
+}) {
+  const { integrations, loading: isLoading, refetch } = useIntegrations();
+  const { colorScheme } = useMantineColorScheme();
+  const [emailProviders, setEmailProviders] = useState<IIntegratedProvider[]>([]);
+  const [smsProvider, setSmsProvider] = useState<IIntegratedProvider[]>([]);
+  const [chatProvider, setChatProvider] = useState<IIntegratedProvider[]>([]);
+  const [pushProvider, setPushProvider] = useState<IIntegratedProvider[]>([]);
```

Comment:
for these providers you don't need to have a state, just calculate them in the `useMemo` hook